File: Flowspire-app/src/app/models/RegisterCustomerRequest.ts
export interface RegisterCustomerRequest {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  phoneNumber: string;
  birthDate?: Date | string;    // Optional, use ISO string or Date object
  gender?: string;              // e.g., 'Male', 'Female', 'NotSpecified'
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
}

File: Flowspire-api/Flowspire.API/Models/RegisterCustomerRequest.cs
using Flowspire.Domain.Enums;

namespace Flowspire.API.Models;

public class RegisterCustomerRequest
{
    public string Email { get; set; }
    public string NameFirst { get; set; }
    public string LastName { get; set; }
    public string Password { get; set; }
    public string PhoneNumber { get; set; }
    public DateTime? BirthDate { get; set; }
    public Gender Gender { get; set; } = Gender.NotSpecified;
    public string? AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? Country { get; set; }
    public string? PostalCode { get; set; }
}
